Data-Science/Kubeflow/Lightweight-Python-Functions/lightweight_python_functions_v2.ipynb

```diff
@@ -0,0 +1,204 @@
+    "def preprocess(\n",
+    "    # An input parameter of type string.\n",
+    "    message: str,\n",
+    "    # Use Output[T] to get a metadata-rich handle to the output artifact\n",
+    "    # of type `Dataset`.\n",
+    "    output_dataset_one: Output[Dataset],\n",
+    "    # A locally accessible filepath for another output artifact of type\n",
+    "    # `Dataset`.\n",
+    "    output_dataset_two_path: OutputPath('Dataset'),\n",
+    "    # A locally accessible filepath for an output parameter of type string.\n",
+    "    output_parameter_path: OutputPath(str),\n",
+    "    # A locally accessible filepath for an output parameter of type bool.\n",
+    "    output_bool_parameter_path: OutputPath(bool),\n",
+    "    # A locally accessible filepath for an output parameter of type dict.\n",
+    "    output_dict_parameter_path: OutputPath(Dict[str, int]),\n",
+    "    # A locally accessible filepath for an output parameter of type list.\n",
+    "    output_list_parameter_path: OutputPath(List[str]),\n",
+    "):\n",
+    "    \"\"\"Dummy preprocessing step.\"\"\"\n",
+    "\n",
+    "    # Use Dataset.path to access a local file path for writing.\n",
+    "    # One can also use Dataset.uri to access the actual URI file path.\n",
+    "    with open(output_dataset_one.path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    # OutputPath is used to just pass the local file path of the output artifact\n",
+    "    # to the function.\n",
+    "    with open(output_dataset_two_path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    with open(output_parameter_path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    with open(output_bool_parameter_path, 'w') as f:\n",
+    "        f.write(\n",
+    "            str(True))  # use either `str()` or `json.dumps()` for bool values.\n",
+    "\n",
+    "    import json\n",
+    "    with open(output_dict_parameter_path, 'w') as f:\n",
+    "        f.write(json.dumps({'A': 1, 'B': 2}))\n",
+    "\n",
+    "    with open(output_list_parameter_path, 'w') as f:\n",
+    "        f.write(json.dumps(['a', 'b', 'c']))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "b2ed7f5f-bda0-40e4-bcb1-78ac3e91dde0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@component(base_image=BASE_IMG, install_kfp_package=False)\n",
+    "def train(\n",
+    "    # Use InputPath to get a locally accessible path for the input artifact\n",
+    "    # of type `Dataset`.\n",
+    "    dataset_one_path: InputPath('Dataset'),\n",
+    "    # Use Input[T] to get a metadata-rich handle to the input artifact\n",
+    "    # of type `Dataset`.\n",
+    "    dataset_two: Input[Dataset],\n",
+    "    # An input parameter of type string.\n",
+    "    message: str,\n",
+    "    # Use Output[T] to get a metadata-rich handle to the output artifact\n",
+    "    # of type `Dataset`.\n",
+    "    model: Output[Model],\n",
```

Comment:
Comment is mentioning `Output[T] ... of type `Dataset`, but formal parameter is defined as `model: Output[Model]`.